### ISSUE TITLE
Cannot run the project from editor if godot was started with --path [FreeBSD]

### ISSUE THREAD

**Issue by robfram** [godotengine/godot#15361]
**Godot version:**
custom-built 8ec06de6681a600bd5a74b4a1bdea85f656b5a88
It failed in previous ones too, as bc513dfa84c2bdfcfbf82c924cbf3fd3bc9c27de

**OS/device including version:**
FreeBSD 11.0-RELEASE-p1 r306420
Intel(R) Core(TM) i3-2330M CPU @ 2.20GHz (2195.06-MHz K8-class CPU)

**Issue description:**
When click on RUN icon from Editor after using --path argument from command line to choose directly the project (skipping the Project Manage GUI), godot can't start the project.

The problem is that godot is not storing the right absolute path when the --path argument is used.

```
 Running: /usr/home/antemil/ocio/dev/godot/projects/pacman/bin --path /usr/home/antemil/ocio/dev/godot/projects/pacman --remote-debug 127.0.0.1:6007 --allow_focus_steal_pid 8440 --position 283,-16
**ERROR** OS_Unix::execute - Could not create child process while executing: /usr/home/antemil/ocio/dev/godot/projects/pacman/bin
```

**Steps to reproduce:**
$ bin/godot.x11.tools.64 --path ../projects/pacman/ --editor

**Comment by Jerome67000** [comment]
It's working fine on Windows.

I see you are on FreeBSD and it is a specific case with it, can someone try to see if this issue happen in other distro too ? @akien-mga maybe related to the fix of #15288 ?

https://github.com/godotengine/godot/blob/8e8240450edc48f4fb379e593a4ef7f54ab656b7/drivers/unix/os_unix.cpp#L301-L311

**Comment by robfram** [comment]
Definitively there is something wrong with path and exec searchs at least in FreeBSD. Yesterday I discover another error related to this. Deleting resources causes to drop the same kind of error text.

For example, creating a brand new test project and deleting the default icon.png resource, the GUI stalls for a few moments and this output appears in the shell, and of course the resource is still there.

```
$ ./godot.x11.tools.64
No touch devices found
W: [(null)] caps.c: Normally all extra capabilities would be dropped now, but that's impossible because PulseAudio was built without capabilities support.
OpenGL ES 3.0 Renderer: Mesa DRI Intel(R) Sandybridge Mobile 
GLES3: max ubo light: 409
GLES3: max ubo reflections: 455, ubo size: 144
ARVR: Registered interface: Native mobile
OPENING: /home/antemil/godot/test (::home::antemil::godot::test)
No touch devices found
OpenGL ES 3.0 Renderer: Mesa DRI Intel(R) Sandybridge Mobile 
GLES3: max ubo light: 409
GLES3: max ubo reflections: 455, ubo size: 144
ARVR: Registered interface: Native mobile
Moving to trash: /usr/home/antemil/godot/test/icon.png
**ERROR** OS_Unix::execute - Could not create child process while executing: mkdir
**ERROR** OS_Unix::execute - Could not create child process while executing: mv
```
All this is at v3.0.beta.custom_build 8ec06de

I'm compiling the 0e6e98a65f26c915b5717992ba881cf33406c8bf commit to confirm it still happens. I would love to provide more debug info, but as mentioned in issue #15367 gdb crashed as well.

**Comment by robfram** [comment]
Is it any reason to have differente exec for FreeBSD? The mkdir/mv error is solved if the FreeBSD branch is removed in drivers/unixi/os_unix.cpp:
```
        Vector<char *> args;
        for (int i = 0; i < cs.size(); i++)
            args.push_back((char *)cs[i].get_data()); // shitty C cast
        args.push_back(0);

        execvp(p_path.utf8().get_data(), &args[0]);

        // still alive? something failed..
        fprintf(stderr, "**ERROR** OS_Unix::execute - Could not create child process while executing: %s\n", p_path.utf8().get_data());
        abort();
```
```
$ bin/godot.x11.tools.64.llvm
No touch devices found
W: [(null)] caps.c: Normally all extra capabilities would be dropped now, but that's impossible because PulseAudio was built without capabilities support.
OpenGL ES 3.0 Renderer: Mesa DRI Intel(R) Sandybridge Mobile 
GLES3: max ubo light: 409
GLES3: max ubo reflections: 455, ubo size: 144
ARVR: Registered interface: Native mobile
OPENING: /home/antemil/godot/projects/test (::home::antemil::godot::projects::test)
No touch devices found
OpenGL ES 3.0 Renderer: Mesa DRI Intel(R) Sandybridge Mobile 
GLES3: max ubo light: 409
GLES3: max ubo reflections: 455, ubo size: 144
ARVR: Registered interface: Native mobile
Duplicating res://ghost_fear.png -> res://ghost_fear2.png
call rescan!
Moving res://ghost_fear2.png -> res://ghost_fear_copy.png
  Remap: res://ghost_fear2.png -> res://ghost_fear_copy.png
call rescan!
Moving to trash: /usr/home/antemil/godot/projects/test/ghost_fear_copy.png
```

**Comment by akien-mga** [comment]
> Is it any reason to have differente exec for FreeBSD?

No idea, it was changed by @sambler in 73ca870c81145865d5a73c7aa95431c6792d5ec1 three years ago.

**Comment by robfram** [comment]
Perhaps it is only an optimization trick to avoid searching the executable in the full path if it is an absolute path, but as FreeBSD has a POSIX implementation of execvp as well as LiNUX does, the trick maybe not mandatory and could be let to do the same thing as LiNUX.

**Comment by sambler** [comment]
Actually I have a better fix to replace that, I just hadn't got around to making the pull request. The fix has been in the ports tree for some time.

**Comment by akien-mga** [comment]
Fixed by #15475.
